packages/components/stories/icon/README.md

```diff
@@ -0,0 +1,138 @@
+
+const App = () => (
+    <>
+        <Icon icon='IcBell' height={32} width={48} /> // icon with different width and height
+        <Icon icon='IcBell' size={128} />   // icon with the same width and height
+        <Icon icon='IcBell' color='active' />
+        <Icon icon='IcBell' customColor='#dedede' />
+    </>
+);
+```
+
+## **Adding new icon**
+1. Add an `.svg` icon file under one of the svg folders (or add a new folder) in `components/icon` folder.
+2. Make sure the icon file name is in  `kebab-case` and is prefix with `ic-`. The icon file name will be converted to a `PascalCase` as the icon name.
+
+    - Example:
+
+    ```sh
+    ic-hello-world.svg --> IcHelloWord
+    ```
+
+    - then you can pass the icon name to render the icon:
+
+    ```jsx
+    <Icon icon='IcHelloWorld' />
+    ```
+
+3. Run `svgo` with the `--enable=sortAttrs` flag to clean and sort the icon attributes (for readibility).
+
+    ```sh
+    svgo src/components/icon/**/**.svg --enable=sortAttrs
+    ```
+
+4. Run `npm run build` to rebuild the Icon component.
+
+    - Note: Running `npm run build` command will run the `npm run build:icons`. This command will then rewrite `icons.js` file and `stories/icon/icons.js` files with the newly added icon imports.
+    - Tips: Run `npm run storybook` to check whether the newly added icon is added.
+
+5. If you happen to have a bunch of svg files to rename, you can use the `utils/rename.js` script to rename them.
+
+    ```sh
+    node rename.js path/to/directory 'string-to-search' 'string-to-replace'
+    ```
+
+## **Theming icon**
+1. Theming has to be done manually for now. You need add CSS variables to each path that needs to be themed.
+2. Rules:
+    - replace primary `<path />` fill or stroke color with `var(--fill-color1)`
+    - replace secondary `<path />` fill or stroke color with `var(--fill-color2)`
+    - replace background `<rect />` or `<circle />` fill color with `var(--fill-color3)`
+
+    Example:
+
+    ```svg
+    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16">
+        <g>
+            <path fill="var(--fill-color1)" d="..." />
+            <path stroke="var(--fill-color2)" d="..." />
+            <rect fill="var(--fill-color3)" d="..." />
+        </g>
+    </svg>
+    ```
+
+## **Adding new SVG sprite**
+1. Configure webpack to create new SVG sprite:
+
+    ```js
+    {
+        loader : 'svg-sprite-loader',
+        options: {
+            extract       : true,
+            spriteFilename: svgPath => {
+                ...
+                if (svgPath.includes('components/icon/new_category')) {
+                    return 'new_category.svg'; // Webpack will add new_category.svg to public/sprite
+                }
+                ...
+            },
+            publicPath: '/icon/sprite/',
+        },
+    },
+    ```
+
+2. Configure Icon component to read from the new SVG sprite:
+
+    ```jsx
+    const Icon = ({
+        ...
+    }) => {
+        ...
+        if (icon) {
```

Comment:
Can we change this to if `(typeof icon === 'string')` instead of the nested if statements? what are the other possible values for this?

Reply:
@reza-fs Sure, we can change it. There's no other values possible for this, the icon name should only be `string` 😄